xLights/sequencer/RowHeading.cpp

```diff
@@ -1636,27 +1639,57 @@ void RowHeading::render( wxPaintEvent& event )
                     }
                 }
 
-                bool hasEffects = rowInfo->element->HasEffects();
-                if (!hasEffects && m->GetDisplayAs() == "ModelGroup")
-                {
-                    // model groups are only marked if model group has direct effects or the model with effects is otherwise hidden in the view
-                    hasEffects = rowInfo->element->HasEffects();
+                bool hasDirectEffectsIndicator = false;
+                bool containsEffectsIndicator = rowInfo->element->HasEffects();
+                hasDirectEffectsIndicator = (rowInfo->element->GetEffectLayer(rowInfo->layerIndex)->GetEffectCount() > 0);
 
+                if (m->GetDisplayAs() == "ModelGroup")
+                {
                     int view = mSequenceElements->GetCurrentView();
                     ModelGroup* mg = dynamic_cast<ModelGroup*>(m);
                     auto models = mg->ModelNames();
-                    for (auto it = models.begin(); !hasEffects && it != models.end(); ++it) {
+                    for (auto it = models.begin(); !containsEffectsIndicator && it != models.end(); ++it)
+                    {
                         ModelElement* mm = dynamic_cast<ModelElement*>(mSequenceElements->GetElement(*it));
 
-                        if (mm != nullptr && !ModelInView(*it, view)) {
```

Comment:
Note, ModelInView() is a painful O(n squared) linear search, that has been removed.